core/src/dird/ua_restore.cc

```diff
@@ -1187,9 +1179,7 @@ static bool BuildDirectoryTree(UaContext* ua, RestoreContext* rx)
   if (tree.FileCount == 0) {
     OK = AskForFileregex(ua, rx);
     if (OK) {
-      last_JobId = 0;
       for (p = rx->JobIds; GetNextJobidFromList(&p, &JobId) > 0;) {
-        if (JobId == last_JobId) { continue; /* eliminate duplicate JobIds */ }
```

Comment:
I removed it at first because it `last_jobid` was always `0`. I thought it was just a leftover from something else.
But now I brought it back and made it do the actual comparison and update the `last_jobid`